api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1111,7 +1111,22 @@ public void setValueAsString(String s) throws ParseException {
 			} else if ("ST".equals(abbrev)) {
 				setValueText(s);
 			} else {
-				throw new RuntimeException("Don't know how to handle " + abbrev);
+				String[] supportedFormats = { "yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS",
+					"yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ssXXX", "yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm:ss"};
+				boolean flag = false;
+				for (int i = 0; i < supportedFormats.length; i++) {
+					try {
+						DateFormat dateFormat = new SimpleDateFormat(supportedFormats[i]);
+						setValueDatetime(dateFormat.parse(s));
+						flag = false;
+						break;
+					}
+					catch (Exception ex) {
+						flag = true;
+					}
+				}
+				if (flag)
+					throw new RuntimeException("Don't know how to handle " + abbrev);
```

Comment:
Why don't do it directly in catch?

Reply:
I am trying to match the date format to any of the allowed date formats specified in the array. if i do it in the catch directly, and say the date being tested does not conform to the first format in the array, the catch will be triggered and in that case the exception will be thrown, even if the date does indeed conform to one of the later specified formats. Hence my use of the flag